core/eval.go

```diff
@@ -87,6 +87,11 @@ func evalSET(args []string) []byte {
 				return Encode(errors.New("ERR value is not an integer or out of range"), false)
 			}
 			exDurationMs = exDurationSec * 1000
+		case "NX", "nx":
+			obj := Get(key);
+			if obj != nil {
+				return Encode(fmt.Errorf("ERR %s key exists", key), false);
```

Comment:
Redis does not raise an error if the object already exists. It simply ignores. it. Stick to how Redis behaves under similar situation.